mantidproject/mantid
# cancel algorithm when cancelling script in workbench

**Description of work.**

Currently, when you stop a script in workbench you have to wait for the current algorithm to finish before it will stop.

This change makes it send a cancel request to the algorithm as well, so the script should stop much sooner.



**To test:**


1. run this script
   ```python
   Pause(10) #pause for 10 seconds
   print('Done')
   ```
1. press Ctrl+D to abort the script. It will stop running after the 10 seconds, not soon after your keyboard interrupt
1. Now repeat this with multiple script tabs open and make sure it cancels the right algorithm.

Fixes #28310

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
